### PR TITLE
Add support for moving focus between panes with the keyboard

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Enables the user to set keybindings to move focus between panes with the keyboard.

 
## References
See #1000 for panes megathread

## PR Checklist
* [x] Closes #955

## Detailed Description of the Pull Request / Additional comments

This is highly based off the work done for resizing panes. Same logic applies - moving focus will move up the panes tree until we find a pane to move the focus to. 

## Validation Steps Performed

Opened a bunch of panes, kept moving focus.

### CODE REVIEW

File: src/cascadia/TerminalApp/Pane.h [hunk 1]
```diff
@@ -82,6 +83,9 @@ class Pane : public std::enable_shared_from_this<Pane>
     void _ApplySplitDefinitions();
     bool _Resize(const winrt::TerminalApp::Direction& direction);
 
+    bool _Resize(const winrt::TerminalApp::Direction& direction);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/Pane.h]: oh no

Comment (miniksa) [reply src/cascadia/TerminalApp/Pane.h]: image

File: src/cascadia/TerminalApp/Pane.cpp [hunk 2]
```diff
@@ -187,6 +187,95 @@ bool Pane::ResizePane(const Direction& direction)
...
+}
+
+// Method Description:
+// - Attempts to move focus to one of our children. If we have a focused child,
+//   we'll try to move the focus in the direction requested.
+//   - If there isn't a pane that exists as a child of this pane in the correct
+//     direction, we'll return false. This will indicate to our parent that they
+//     should try and move the focus themselves. In this way, the focus can move
+//     up and down the tree to the correct pane.
+// - This method is _very_ similar to ResizePane. Both are trying to fine the
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]:
```suggestion
// - This method is _very_ similar to ResizePane. Both are trying to find the
```

File: src/cascadia/TerminalApp/Pane.cpp [hunk 3]
```diff
@@ -187,6 +187,95 @@ bool Pane::ResizePane(const Direction& direction)
...
+        if ((!_firstChild->_IsLeaf()) && _firstChild->_HasFocusedChild())
+        {
+            return _firstChild->NavigateFocus(direction) || _NavigateFocus(direction);
+        }
+        else if ((!_secondChild->_IsLeaf()) && _secondChild->_HasFocusedChild())
+        {
+            return _secondChild->NavigateFocus(direction) || _NavigateFocus(direction);
+        }
+    }
+    return false;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]: I'm worried that there's a `_NavigateFocus` in three of the .. four? states for this conditional. Is there a better way to structure this code?

**Comment by mdtauk** [comment]
Are these panes being implemented as XAML controls, or something more Win32 based?

File: src/cascadia/TerminalApp/Pane.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp]: (we discussed on chat that this is fine the way it is an there's not really a better structure for it)

**Pull request merged**